install/kubernetes/istio-gateway.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-gateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default controller
```

Comment:
Please don't overload the ingress gateway. This is what end users will use for their ingress configuration. Is it possible to launch a separate gateway that serves as the ILB instead of exposing pilot to the outside world?

Reply:
Let's hear what everybody has to say about this, and we can come to a consensus.

 As far as I'm concerned, this is really a doc issue because we don't have to dictate how people deploy it in their environment. Even if we have this PR completed and merged, it's just one way to deploy it.

Reply:
Yes, but our doc is stating move from kube ingress and use istio ingress gateway. and every non-expert user so far (if you monitor the mailing lists) seems to use the default gateway we ship with `istio: ingressgateway`. By shipping a separate clusterGateaway, its very clear to people what it does and allows people to enable/disable one or more gateways. If this gateway does not work, it won't impact the common use case of using it for ingress. 
Implicitly overloading a resource that is solely meant for the end user is a bad idea. Its not a doc issue at this point.

Reply:
Sorry that I didn't mean to say that using ingress gateway or a new gateway for this purpose is a doc issue. I meant to say that the original issue this PR is trying to address is mainly a doc issue. what matters to this PR is what we want to ship with istio as one way to deploy the istio multi-cluster control plane. So let's come to some kind of agreement on that. Basically, it comes down to these few things:
 -- Use a separate istio gateway workload for the istio control components, or use the existing ingress gateway (which you don't favor). I have no preference. 
 -- the list of services/ports to expose.
 -- integrated with helm
 -- how to deal with UDP. We may end up having this gateway not being able to cover all the istio control components.

Reply:
I have no preference on anything except for the first one, where I prefer using separate istio gateway workload for multi-cluster stuff. To keep things isolated. as for others, I will let others pitch in

Reply:
@rshriram so just to be clear, we could add something like "managementgateway" which is actually an ingressgateway.  Then this new component (whatever its called) could take the general ingressgateway rules and manage them?  How would these two ingress servers selectively choose which rules they are to honor?

As an aside, We have other services in Istio (addons) which demand load balancers at startup.  We could integrate those into this managementgateway as well for future work.

Thoughts?

Reply:
Have separate Gateway - yes.
Different deployment - please don't. IPs and LBs are expensive and have quotas.

Reply:
@costinm if we do not have a different deployment, then all of the load will be handled by same deployment, this will cause the workload and the multicluster will impact each other. 

As multicluster gateway is a specified gateway for multicluster support, how about use a separate deployment to handle this?